### PR TITLE
Settings HIL test

### PR DESCRIPTION
**Pull request by sam-golioth**
This adds a new HIL test for the Settings service. This is part of our "comprehensive" HIL tests and so includes error and edge cases. If you see anything that you think should be covered by this test, but isn't, please leave a comment.

This also updates our CI workflows to automatically detect, build, and run new HIL tests as they are added.

Resolves golioth/firmware-issue-tracker#320

### CODE REVIEW

File: .github/workflows/hil_test_esp-idf.yml [hunk 1]
```diff
@@ -54,13 +60,13 @@ jobs:
         run: |
           pip install pytest pytest-timeout
           pip install tests/hil/scripts/pytest-hil
-          pip install git+https://github.com/golioth/python-golioth-tools@v0.5.1
+          pip install git+https://github.com/golioth/python-golioth-tools@f6bd62aaf239409c57b2a0ccbb65c2df0833564f
```
Comment (sam-golioth) [inline .github/workflows/hil_test_esp-idf.yml:63]: Point to `v0.6.0` after https://github.com/golioth/python-golioth-tools/pull/29 is merged

File: .github/workflows/hil_test_zephyr.yml [hunk 2]
```diff
@@ -81,13 +85,13 @@ jobs:
         run: |
           pip install pytest pytest-timeout
           pip install tests/hil/scripts/pytest-hil
-          pip install git+https://github.com/golioth/python-golioth-tools@v0.5.1
+          pip install git+https://github.com/golioth/python-golioth-tools@f6bd62aaf239409c57b2a0ccbb65c2df0833564f
```
Comment (sam-golioth) [inline .github/workflows/hil_test_zephyr.yml:88]: Point to v0.6.0 after https://github.com/golioth/python-golioth-tools/pull/29 is merged

**Review by mniestroj — APPROVED** [review]
